src/rust/ensogl/src/display/shape/primitive/glsl/fragment_runner.glsl

```diff
@@ -1,21 +1,26 @@
 /// This code is the body of the fragment shader main function of a GLSL shape.
 
-Env   env      = Env(1);
-vec2  position = input_local.xy ;
-Shape shape    = run(env,position);
-float alpha  = shape.color.color.raw.a;
+Env   env        = Env(1);
+vec2  position   = input_local.xy ;
+Shape shape      = run(env,position);
+float alpha      = shape.color.color.raw.a;
+uint alpha_no_aa = alpha > 0.5 ? uint(1) : uint(0);
+
+output_id = uvec4(input_symbol_id,input_instance_id,0,alpha_no_aa);
+output_id.r *= alpha_no_aa;
+output_id.g *= alpha_no_aa;
+output_id.b *= alpha_no_aa;
 
 if (input_display_mode == 0) {
     output_color = srgba(unpremultiply(shape.color)).raw;
     output_color.rgb *= alpha;
 } else if (input_display_mode == 1) {
     Rgb col = distance_meter(shape.sdf.distance, 200.0 * input_zoom * input_pixel_ratio, 200.0/input_zoom * input_pixel_ratio);
     output_color = rgba(col).raw;
+}else if (input_display_mode == 2) {
+    float r = float(((int(input_symbol_id) * 79) % 360)) / 360.0;
+    float g = float(((int(input_instance_id) * 43) % 360)) / 360.0;
+    float b = float(((int(input_instance_id) * 97) % 360)) / 360.0;
```

Comment:
I'm not convinced this is the best idea for this debug mode. When we've been chatting on Discord I provided you with `hsv2rgb` function and sample implementation on purpose (and my bad I did not explain it good enough). There are 2 important things when using HSV translation instead:

1. We can still get every ID visually distinct, but IDs close to each other have similar colors, and this is actually very helpful information sometimes.
2. We can sample the color on screen, put it in any RGB to HSV converter and based on the hue easily compute what was the original ID we are sampling. It is not very easy by using your equation here.

Reply:
Ah, I didn't know this was supposed to be invertible.
This has a similar feature: shapes with the same `input_instance_id` will look more similar than shapes with a different `input_instance_id`.
I'll do something invertible then.